tests/test_stubs.py

```diff
@@ -659,6 +687,34 @@ def test_render_typed_dict_in_list(self):
         self.maxDiff = None
         assert build_module_stubs(entries)['tests.util'].render() == expected
 
+    def test_render_typed_dict_base_and_subclass(self):
+        function = FunctionDefinition.from_callable_and_traced_types(
+            Dummy.an_instance_method,
+            {
+                'foo': make_typed_dict(required_fields={'a': int}, non_required_fields={'b': str}),
+                'bar': int,
+            },
+            int,
+            None,
+            existing_annotation_strategy=ExistingAnnotationStrategy.IGNORE,
+        )
+        entries = [function]
+        expected = '\n'.join([
+            'from mypy_extensions import TypedDict',
+            '',
+            '',
+            f'class FooNonTotalTypedDict({make_forward_ref("FooTypedDict")}, total=False):',
+            '    b: str',
+            '',
+            '',
+            'class FooTypedDict(TypedDict):',
+            '    a: int',
```

Comment:
Why do we choose to render these in this order, requiring the ForwardRef, when we could just render them in the other order instead?

Reply:
We render them in alphabetical order. The only reason I went for the ForwardRef was because it seemed pretty brittle to have it work only because the base class name was < the subclass name. 

Let me know if you're ok with `FooTypedDict` and `FooTypedDictNonTotal` so that they are in alphabetical order.

Reply:
Yeah that seems fine.